### PR TITLE
refactor(runtime): canonicalize legacy BMI visualization access

### PR DESCRIPTION
**Pull request by Katsiarynakavaleuskaya**
# Pull Request

## Summary

Canonicalize the hidden legacy `/bmi` visualization provider without changing BMI math, renderer behavior, routes, schemas, auth, OpenAPI, generated clients, or application identity.

- Remove `sys.modules`, `resolve_attr`, and mutable `app` / `legacy_app` facade precedence from `app.services.bmi_compat`.
- Use only the service-local `MATPLOTLIB_AVAILABLE` and `generate_bmi_visualization` bindings while preserving existing truthy `available` semantics and exact unavailable payloads.
- Replace false facade-authority and nonexistent-route coverage with deterministic consumer-binding, route-visibility, error-hygiene, and non-authority contracts.
- Reconcile the legacy compatibility map and backlog ledger; PRO nutrition reverse-dependency decomposition remains the next separate lane.

- [x] I reviewed `docs/ENGINEERING_LESSONS.md` and followed repo policies.
- [x] Select one change type:
  - [ ] Bug fix
  - [ ] Feature
  - [x] Refactor
  - [ ] Docs
- [x] Linked PRs: #2140 and #2141 are merged prerequisites. #2138/#2139 remain separate overview/guard lanes and are not modified here.

## Lane Start Provenance

Packet: `artifacts/orchestration/task_packets/15273f7aa1e0.json`

Local packet, gitignored.

- Pre-open role order: `agent-coordinator -> architecture-specialist -> backend-engineer -> security-auditor -> qa-engineer-agent -> bug-hunter`.
- Post-open order: `qa-engineer-agent -> bug-hunter -> security-auditor -> one Codex Security material-diff scan -> pulseplate-pr-review`.
- Operator override: continue this BMI lane while the separate automatic-review governance task remains out of scope. This PR does not edit bot or review configuration.

## Risk & Impact

- [ ] User-facing change
- [ ] Data model/migration
- [x] Security-sensitive
- [ ] Performance-sensitive

The public contract is unchanged. The security-relevant outcome is narrower runtime authority: mutable facade rebinding can no longer select the renderer, and renderer-internal failure details remain excluded from the normalized compatibility response.

## Scope

- One production service cutover to direct local renderer bindings.
- Exact tests for short-circuiting, renderer arguments, truthy payload preservation, failure normalization, facade non-authority, and route/OpenAPI visibility.
- Removal or retargeting of tests that encoded the deleted resolver or a nonexistent `/api/v1/bmi/visualize` route.
- Compatibility ownership and backlog truth updates.

## Out Of Scope

- `bmi_visualization.py` renderer implementation and matplotlib dependency policy.
- Structured `BMIScaleV1Spec` visualization.
- BMI formulas, thresholds, schemas, routers, auth, or public API.
- `legacy_app.py` / `app.__init__` compatibility-export removal.
- PRO nutrition, insight, app-factory inversion, and the large legacy-growth guard.
- Automatic-review governance owned by the parallel lane.

## Tests

- [x] Unit tests updated/added
- [x] Integration/route tests updated
- [x] Manual contract verification completed

Local evidence through `be9041cd69f339ceb933cf28ff2f0cbfbe677bcd`:

- `python3 scripts/orchestration/check_preflight.py` - PASS.
- `python3 scripts/orchestration/check_agent_consistency.py` - PASS.
- Focused changed-test bundle - PASS (289 tests).
- Route/public/OpenAPI/security oracle bundle - PASS.
- `python3 scripts/ci/check_legacy_growth_guard.py` and `tests/test_legacy_growth_guard.py` - PASS.
- Targeted MyPy for `app/services/bmi_compat.py` - PASS.
- `make openapi-check` - PASS; generated OpenAPI/TypeScript artifacts have zero diff.
- `make validate-changed` - PASS; the selector chose all 13 changed Python test suites.
- `pre-commit run --all-files` - PASS.
- Pre-push MyPy, pip-audit, backend tests, full-repo Bandit, and Docker build test - PASS.
- `git diff --check` - PASS.

Local full `make verify` was not run, per the repository machine-budget rule. Current-head GitHub CI remains mandatory before merge readiness.

## Experiment Runner Evidence

Not applicable: Experiment Runner did not materially contribute to this closeout commit; the retained accepted Apple Container artifact remains preserved locally.

Apple Container 1.1.0 strict oracle execution was accepted with three immutable oracles at return code 0, `network_budget=0`, `shared_tree_untouched=true`, `mutated_paths=[]`, and image digest `sha256:8b95aa8a94d989ff18af7449fbb0feae6783623a7bf49434f0e16341bd61c483`.

Attribution: Not applicable. The runner confirmed an already stabilized decision and did not materially shape a commit.

## CI Gates

- [x] Required current-head checks terminal green
- [x] Diff coverage >=97%
- [x] Applicable security/governance checks terminal

## Split Justification

Production behavior changes in one service file. The additional files delete or retarget tests that encode the removed facade authority or a nonexistent route. Splitting would leave false contract evidence against the same runtime cutover.

## Discussion Thread Pass

- [x] Discussion-thread pass completed
- [x] Fixed in commit mapping completed

### Fixed in Commit Mapping

Canonical mapping: [Canonical review mapping](https://github.com/Katsiarynakavaleuskaya/PulsePlate/blob/6a0a3e68f2cf9c5495ee340a7d236c38bc5aeec9/docs/review/PR_2145_FIXED_MAPPING.md)

Only real GitHub threads and reachable post-comment commits are mapped; no synthetic or unavailable execution reference is assigned a commit SHA.

## Review / Scan Contract

Completed for frozen material digest `sha256:2231f076a4bb1c9c609781af66ab4c9af76a125c23955cb8840abc6cac88c1e1`:

- exact-head Codex review: `https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/2145#issuecomment-5011870818`;
- completed Codex Security scan: `f93bcd78-06d9-4506-90db-7288a3ceee58`;
- canonical QA, bug-hunter, and security-auditor passes were completed before freeze.

No new scan was launched for closeout. The unchanged exact-head scan was admitted through a deterministic local V1 compatibility projection; the original scan directory remains untouched. Mapping/body-only closeout does not reopen material review.

## Merge Readiness (Mandatory)

- [x] All required checks are green on the latest commit
- [x] No unresolved review threads
- [x] No actionable bot comments remain undispositioned
- [x] Strict authenticated merge-readiness passes
- [x] Mandatory wait-window completed

## Deferred / Follow-ups

- [x] `docs/roadmap/BACKLOG_LEDGER.md` names PRO nutrition reverse-dependency decomposition as the next bounded legacy lane.
- [ ] GitHub issue(s): none required for this slice.

## Notes

- No database, schema, dependency, workflow, route, auth, or generated-client change.
- No new feature flag or environment variable.
- Rollback is a normal revert; no data rollback is required.
- Cleanup after confirmed merge must remove only this branch and disposable local caches/artifacts. Explicitly retained creative/Experiment Runner evidence is preserved.

---
## Summary by cubic
Canonicalized legacy BMI visualization access in the hidden `/bmi` flow to use only service-local bindings. Public behavior and APIs remain unchanged.

- **Refactors**
  - Simplified `app/services/bmi_compat` to call local `generate_bmi_visualization` and `MATPLOTLIB_AVAILABLE` only; removed `sys`/facade/module-table resolution and `resolve_attr`.
  - Skips work when `include_chart` is false; if matplotlib is missing, returns a consistent unavailable payload; preserves truthy `available`; normalizes failures to a generic error.
  - Locked `/bmi` to service-local bindings; ensured `/api/v1/bmi` never calls the legacy renderer.

- **Tests/Docs**
  - Added AST guard to forbid `sys`/`legacy_app`/`resolve_attr` and string-literal/module-table references; subprocess clean-import proof (`TESTING=true`, timeout); `application/json` header checks.
  - Asserted `/bmi` uses service bindings even if facades are patched; args forwarded; generator not invoked when `include_chart` is false or matplotlib unavailable; internal errors don’t leak; `/api/v1/bmi` never calls legacy viz.
  - Removed/retargeted tests tied to facade authority or `/api/v1/bmi/visualize`; updated `docs/architecture/LEGACY_COMPATIBILITY_SEAM.md`; added the canonical review seal; minimal `.secrets.baseline` line updates.

<sup>Written for commit 6a0a3e68f2cf9c5495ee340a7d236c38bc5aeec9. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/Katsiarynakavaleuskaya/PulsePlate/pull/2145?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved BMI visualization behavior for the compatibility route: consistent unavailable vs generation-failure responses, honors `include_chart`, and avoids leaking internal error details.
  * Added clearer edge-case validation: PDF export returns a 503 when required `to_pdf_day` is missing/invalid; unknown micronutrient keys now error correctly.

* **Documentation**
  * Clarified legacy BMI visualization compatibility ownership boundaries and updated the roadmap/backlog notes.

* **Tests**
  * Updated and expanded endpoint and service coverage for import-order stability, scheduler compatibility, and visualization/validation edge cases.

### CODE REVIEW

**Comment by Katsiarynakavaleuskaya** [comment]
@codex review

**Comment by Katsiarynakavaleuskaya** [comment]
@codex review

**Comment by Katsiarynakavaleuskaya** [comment]
@codex review

Final material head: `327104789481297ad87d2fb04b1cc855ea975a49`
Material digest: `sha256:2231f076a4bb1c9c609781af66ab4c9af76a125c23955cb8840abc6cac88c1e1`

Please review this exact material diff. This request supersedes the ancestor-head review only because the material test diff changed to address live CodeRabbit findings.

File: docs/review/PR_2145_FIXED_MAPPING.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+## Discussion Thread Pass
+- [x] Discussion-thread pass completed
+- [x] Fixed in commit mapping completed
+
+## Fixed in Commit Mapping
+
+Disposition: FIXED
+Commit: 327104789481297ad87d2fb04b1cc855ea975a49
+Evidence: tests/test_bmi_compat_router.py clean-import subprocess now receives TESTING=true and a bounded timeout; focused pytest, make validate-changed, pre-commit and pre-push hooks passed.
+- https://github.com/Katsiarynakavaleuskaya/PulsePlate/pull/2145#discussion_r3608578924 -> 327104789481297ad87d2fb04b1cc855ea975a49
```
[the remark this answers is not in the document]
Comment (Katsiarynakavaleuskaya) [reply docs/review/PR_2145_FIXED_MAPPING.md:20]: Disposition: NOT-A-BUG

Evidence: GitHub compare `327104789481297ad87d2fb04b1cc855ea975a49...6a0a3e68f2cf9c5495ee340a7d236c38bc5aeec9` reports `ahead` with merge base `327104789481297ad87d2fb04b1cc855ea975a49`; the GitHub Commit API returns HTTP 422 for reviewer execution ref `a4050449c409ac6106ac08eb4559a6613a4f02da`.

Reason: the mapped FIX is reachable from the live PR head; the unavailable execution ref is not repository ancestry evidence. Canonical fingerprint: `sha256:4f5c5efe245666f104018da362e95791ba73ac04bd8a114cd94b60f20ea2c15d`.

Mapping: https://github.com/Katsiarynakavaleuskaya/PulsePlate/blob/6a0a3e68f2cf9c5495ee340a7d236c38bc5aeec9/docs/review/PR_2145_FIXED_MAPPING.md
